include/grpc++/server_context.h

```diff
@@ -116,6 +117,8 @@ class ServerContext {
 
   std::shared_ptr<const AuthContext> auth_context() const;
 
+  const census_context* get_census_context() const;
```

Comment:
Without it we've got an inconsistent style, and I'd favor census_context()
over get_census_context() because that's where the rest of the code is.

Can you qualify the name better maybe? const ::census_context
*census_context()... const struct census_context *census_context()...
surely there's some way to coax even GCC into accepting C++ :)



On Wed, Jul 29, 2015 at 10:51 AM Alistair Veitch <notifications@github.com>
wrote:

> In include/grpc++/server_context.h
> <https://github.com/grpc/grpc/pull/2670#discussion_r35789668>:
>
> > @@ -116,6 +117,8 @@ class ServerContext {
> >
> >    std::shared_ptr<const AuthContext> auth_context() const;
> >
> > +  const census_context* get_census_context() const;
>
> That results in errors like:
>
> In file included from include/grpc++/stream.h:40:0,
> from include/grpc++/generic_stub.h:38,
> from src/cpp/client/generic_stub.cc:34:
> include/grpc++/server_context.h:120:42: error: declaration of ‘const
> census_context* grpc::ServerContext::census_context() const’ [-fpermissive]
> const census_context* census_context() const;
> ^
> In file included from include/grpc++/impl/call.h:38:0,
> from include/grpc++/channel_interface.h:40,
> from include/grpc++/stream.h:37,
> from include/grpc++/generic_stub.h:38,
> from src/cpp/client/generic_stub.cc:34:
> include/grpc++/client_context.h:51:8: error: changes meaning of
> ‘census_context’ from ‘struct census_context’ [-fpermissive]
> struct census_context;
> ^
> make: *** [objs/opt/src/cpp/client/generic_stub.o] Error 1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/2670/files#r35789668>.
>